### ISSUE TITLE
When I lower the set-point, PID goes to min

### ISSUE THREAD

**Issue by lukepighetti** [soloam/ha-pid-controller#31]
Temp was 275f, set point was 275f. `P` is `3` and `I` is `0.01`

Changed set point to 250f. It immediately dropped to the minimum. I expected it to smoothly transition. 

Is this a tuning problem or a bug? 

<img width="245" alt="Screen Shot 2022-10-18 at 12 57 32 PM" src="https://user-images.githubusercontent.com/666539/196496404-23c4e462-737b-4ba0-9ccd-063c8ba61c36.png">

**Comment by soloam** [comment]
What version of the PID are you running? Is it the latest?

**Comment by soloam** [comment]
Normally that bump is because you might have a very high "p". Give me your config, aren't you using the "d"?

**Comment by Pukinn** [comment]
I have the same issue. But it's not because of a high Kp. It seems the controller resets after changing the set_point. The attributes p, i, d are going to 0. If a high Kp were the reason, I think p should go to a negative value.  
Also it goes to min value, if I increase the set_point (invert == false).

I use the latest version 1.1.5

**Comment by soloam** [comment]
Yes the pid is resetting when the set point is changed! Probably this could
be avoided, but I don't know if in some cases it wouldn't be desirable!

**Comment by Pukinn** [comment]
In which scenario this could be desirable? After a change of the set point I expect a correction because of the derivative part in the next cycle.  
But I increase the set_point and the correction goes down to 0 until the next update of the source.

I think the controller should only reset if the reset is manually triggered.

**Comment by soloam** [comment]
I agree with that! The reason to do this is to avoid sudden spikes up in the pid, but this is making the spike down! I'll fix it!
